src/components/DataTable/TableToolbarDownload.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file Table toolbar button for downloading table data as CSV
+ * @copyright IBM Security 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import Download16 from '@carbon/icons-react/lib/download/16';
+import { ExportToCsv } from 'export-to-csv';
+import IconButton from '../IconButton';
+
+const options = {
+  fieldSeparator: ',',
+  quoteStrings: '"',
+  decimalSeparator: '.',
+  showLabels: true,
+  useTextFile: false,
+  useBom: true,
+};
+
+const TableToolbarDownload = ({ headers, rows, title, filename }) => {
+  const csvRows = rows.map(row => {
+    const newRow = {};
+    headers.forEach(header => {
+      newRow[header.key] = row[header.key];
+    });
+    return newRow;
+  });
+
+  if (title) {
+    options.showTitle = true;
+    options.title = title;
+  }
+  options.filename = filename;
+
+  options.headers = headers.map(head => head.header);
+
+  const csvExporter = new ExportToCsv(options);
+
+  return (
+    <IconButton
+      size="lg"
+      renderIcon={Download16}
+      onClick={() => {
+        csvExporter.generateCsv(csvRows);
+      }}
+    />
+  );
+};
+
+TableToolbarDownload.propTypes = {
+  /**
+   * @type {Object.<Object, *>}
+   * The `rows` prop is where you provide us with a list of all the rows that
+   * you want to render in the table. The only hard requirement is that this
+   * is an array of objects, and that each object has a unique `id` field
+   * available on it.
+   */
+  rows: PropTypes.arrayOf(
```

Comment:
- [ ] We could use `DataTable.propTypes.rows` and `DataTable.propTypes.headers` here to avoid updating these should those change

Reply:
@SimonFinney I have tried many different iterations none of which work when exported and used in a consuming app. I have tried things such as 
```
import { propTypes } from './DataTable';
```
with the usage as:
 ```
rows: propTypes.rows.isRequired,
headers: propTypes.headers.isRequired,
``` 
and receive the error:
```
TypeError: Cannot read property 'rows' of undefined
```
or,
```
import { DataTable } from './DataTable';
```
with the usage as:
```
  rows: DataTable.propTypes.rows.isRequired,
  headers: DataTable.propTypes.headers.isRequired
```
and receive the error:
```
TypeError: Cannot read property 'propTypes' of undefined
```

To note, the package builds fine without any issues.

Reply:
That's odd! This usage sould typically work -

```js
  rows: DataTable.propTypes.rows.isRequired,
  headers: DataTable.propTypes.headers.isRequired
```

I think this is a nice-to-have but probably not a dealbreaker for now - let me know when this ready for re-review!